### PR TITLE
[inductor][cudagraphs] Refactor static_input_id helpers

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #102203
* #102233
* __->__ #102226
* #102123
* #102022

No need of a test case. This functionality is tested pretty thoroughly by existing cases. For rng tests, they will be tested in next PR because it needs other stuff, which needs more discussion.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

### CODE REVIEW

File: torch/_inductor/compile_fx.py [hunk 1]
```diff
@@ -270,7 +273,7 @@ def compile_fx_inner(
         graph = GraphLowering(
             gm,
             shape_env=shape_env,
-            num_static_inputs=num_fixed,
+            num_static_inputs=len(static_input_idxs),
```
Comment (eellison) [inline torch/_inductor/compile_fx.py:276]: I dont think we're adjusting the static inputs for GraphLowering correctly

Comment (anijain2305) [reply torch/_inductor/compile_fx.py]: Good catch! Surprised that none of the tests failed. Pushed a new commit to account for this comment.

File: torch/_inductor/compile_fx.py [hunk 2]
```diff
@@ -555,24 +556,53 @@ def run(new_inputs):
...
-def count_tangents(fx_g: torch.fx.GraphModule):
-    """
-    Infers which inputs are static for a backwards graph
-    """
+def get_static_input_idxs_for_bw(fx_g: torch.fx.GraphModule):
+    # This function returns the range of ids pointing to the location of parameters in the bwd graph
+    # The input signature for the bwd pass from AOT Autograd is
+    #   (*saved_tensors, *tangents)
+    # When functionalization of rng ops is turned on, we have two more inputs
+    #   (seed, offset, *saved_tensors, *tangents)
```
Comment (eellison) [inline torch/_inductor/compile_fx.py:564]: Would it be possible to do `(*saved_tensors, seed, offset, *tangents)` ? That seems a little simpler perhaps ?

File: torch/_inductor/compile_fx.py [hunk 3]
```diff
@@ -555,24 +556,53 @@ def run(new_inputs):
...
+    assert static_arg_idxs == list(out)
+    return out
+
+
+def get_static_input_idxs_for_fw(num_primary_inputs: int, num_aot_graph_inputs: int):
+    # This function returns the range of ids pointing to the location of parameters in the fwd pass
+    # The input signature for the fwd pass from AOT Autograd is
+    #   (*parameters, *primary_inputs)
+    # When functionalization of rng ops is turned on, we have two more inputs
+    #   (seed, offset, *parameters, *primary_inputs)
```
Comment (eellison) [inline torch/_inductor/compile_fx.py:596]: Can we do `(parameters, seed, offset, *primary_inputs)` ?

**Review by eellison — COMMENTED** [review]
Would it be possible to put the seed and offset after the flat parameters and buffers ? This wouldnt require as many changes.

We also have code in aot_autograd which exposes an api to describe params and buffers. And theres logic there that i dont think is being updated, see https://github.com/pytorch/pytorch/blob/main/torch/_functorch/aot_autograd.py#L2371-L2375 and https://github.com/pytorch/pytorch/blob/main/torch/_functorch/aot_autograd.py#L3092-L3095

**Comment by anijain2305** [comment]
Both https://github.com/pytorch/pytorch/blob/main/torch/_functorch/aot_autograd.py#L2371-L2375 and https://github.com/pytorch/pytorch/blob/main/torch/_functorch/aot_autograd.py#L3092-L3095 are still respected.

These are the calling conventions for the runtime_fn of `aot_autograd` return object. The `seed` and `offset` are not visible to this runtime_fn. Instead, they are visible in the fwd and bwd graphs.

**Comment by anijain2305** [comment]
Closing in favor of https://github.com/pytorch/pytorch/pull/102344

**Pull request closed without merging**